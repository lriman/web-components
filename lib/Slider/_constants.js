"use strict";Object.defineProperty(exports,"__esModule",{value:true});exports.REFS=exports.COMPONENTS=void 0;var COMPONENTS={GENERAL:'Slider'};exports.COMPONENTS=COMPONENTS;var REFS={TYPES:{NORMAL:'normal',INVERSE:'inverse'}};exports.REFS=REFS;