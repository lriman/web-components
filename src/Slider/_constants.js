export const COMPONENTS = {
  GENERAL: 'Slider',
};

export const REFS = {
  TYPES: {
    NORMAL: 'normal',
    INVERSE: 'inverse',
  },
};
