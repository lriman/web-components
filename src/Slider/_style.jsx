import styled from 'styled-components';
import {
  getHandleBackgroundColor,
  getHandleBorderColor,
  getRailBackgroundColor,
  getSliderDotStyles,
  getSliderMarkColor,
  getTooltipColor,
  getTrackBackgroundColor,
} from './utils';

export const StyledSlider = styled.div`
  position: relative;
  box-sizing: border-box;
`;

export const StyledSliderContent = styled.div`
  position: relative;
  box-sizing: border-box;
  margin-bottom: 4px;
`;

export const StyledSliderField = styled.div`
  position: relative;
  box-sizing: border-box;
  padding: 0 10px ${props => (props.isWithMarks ? '16px' : 0)} 10px;
`;

export const StyledSliderLite = styled.div`
  .rc-slider {
    position: relative;
    box-sizing: border-box;
    height: 20px;
    padding: 9px 0;
    width: 100%;
    -ms-touch-action: none;
    touch-action: none;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  }

  .rc-slider * {
    box-sizing: border-box;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  }

  .rc-slider-rail {
    position: absolute;
    bottom: 7px;
    width: 100%;
    background-color: ${({ sType, theme }) => getRailBackgroundColor(sType, theme)};
    height: 2px;
  }

  .rc-slider-track {
    position: absolute;
    left: 0;
    bottom: ${({ wideTrack }) => (wideTrack ? '4px' : '6px')};
    height: ${({ wideTrack }) => (wideTrack ? '8px' : '4px')};
    background-color: ${({ sType, theme }) => getTrackBackgroundColor(sType, theme)};
    border-radius: 4px;
    z-index: 1;
  }

  .rc-slider-handle {
    position: absolute;
    margin-left: 0;
    margin-top: -14px;
    width: 32px;
    height: 32px;
    cursor: pointer;
    border-radius: 50%;
    border: solid 8px ${({ sType, theme }) => getHandleBorderColor(sType, theme)};
    background-color: ${({ sType, theme }) => getHandleBackgroundColor(sType, theme)};
    -ms-touch-action: pan-x;
    touch-action: pan-x;
    transition: background-color 0.32s ease-out;
    z-index: 2;
  }

  .rc-slider-handle:focus {
    border: solid 8px ${({ sType, theme }) => getHandleBorderColor(sType, theme, true)};
    outline: none;
  }

  .rc-slider-handle-click-focused:focus {
    background-color: ${({ sType, theme }) => getHandleBackgroundColor(sType, theme, true, false)};
  }

  .rc-slider-handle:hover {
    border: solid 8px ${({ sType, theme }) => getHandleBorderColor(sType, theme, false, true)};
    background-color: ${({ sType, theme }) => getHandleBackgroundColor(sType, theme, false, true)};
  }

  .rc-slider-handle:active {
    cursor: pointer;
  }

  .rc-slider-mark {
    position: absolute;
    bottom: ${({ topMark }) => (topMark ? '45px' : '-13px')};
    left: 0;
    width: 100%;
    font-size: 13px;
    line-height: 16px;
  }

  .rc-slider-mark-text {
    position: absolute;
    display: block;
    vertical-align: middle;
    text-align: center;
    cursor: pointer;
    color: ${({ sType, theme, topMark }) => getSliderMarkColor(sType, theme, topMark)};
    white-space: nowrap;
  }

  .rc-slider-step {
    position: absolute;
    top: 8px;
    width: 100%;
    height: 6px;
    background: transparent;
  }

  .rc-slider-dot {
    ${({ sType, topMark, theme }) => getSliderDotStyles(sType, topMark, theme)}
  }

  .rc-slider-disabled {
    opacity: 0.5;
  }

  .rc-slider-disabled .rc-slider-handle {
    cursor: not-allowed !important;
  }

  .rc-slider-disabled .rc-slider-mark-text {
    cursor: default !important;
  }

  .rc-slider.m-compact .rc-slider-dot {
    opacity: 0;
  }
  .rc-slider.m-compact .rc-slider-dot:first-child,
  .rc-slider.m-compact .rc-slider-dot:last-child {
    opacity: 1;
  }

  .rc-slider.m-compact .rc-slider-mark-text {
    opacity: 0;
  }

  .rc-slider.m-compact .rc-slider-mark-text:first-child,
  .rc-slider.m-compact .rc-slider-mark-text:last-child {
    opacity: 1;
  }

  @media (max-width: 768px) {
    .rc-slider-mark-text {
      opacity: 0;
    }
    .rc-slider-mark-text:first-child,
    .rc-slider-mark-text:last-child {
      opacity: 1;
    }
  }

  .rc-slider-tooltip-zoom-down-enter,
  .rc-slider-tooltip-zoom-down-appear {
    animation-duration: 0.3s;
    animation-fill-mode: both;
    display: block !important;
    animation-play-state: paused;
  }
  .rc-slider-tooltip-zoom-down-leave {
    animation-duration: 0.3s;
    animation-fill-mode: both;
    display: block !important;
    animation-play-state: paused;
  }
  .rc-slider-tooltip-zoom-down-enter.rc-slider-tooltip-zoom-down-enter-active,
  .rc-slider-tooltip-zoom-down-appear.rc-slider-tooltip-zoom-down-appear-active {
    animation-name: rcSliderTooltipZoomDownIn;
    animation-play-state: running;
  }
  .rc-slider-tooltip-zoom-down-leave.rc-slider-tooltip-zoom-down-leave-active {
    animation-name: rcSliderTooltipZoomDownOut;
    animation-play-state: running;
  }
  .rc-slider-tooltip-zoom-down-enter,
  .rc-slider-tooltip-zoom-down-appear {
    transform: scale(0, 0);
    animation-timing-function: cubic-bezier(0.23, 1, 0.32, 1);
  }
  .rc-slider-tooltip-zoom-down-leave {
    animation-timing-function: cubic-bezier(0.755, 0.05, 0.855, 0.06);
  }
  @keyframes rcSliderTooltipZoomDownIn {
    0% {
      opacity: 0;
      transform-origin: 50% 100%;
      transform: scale(0, 0);
    }
    100% {
      transform-origin: 50% 100%;
      transform: scale(1, 1);
    }
  }
  @keyframes rcSliderTooltipZoomDownOut {
    0% {
      transform-origin: 50% 100%;
      transform: scale(1, 1);
    }
    100% {
      opacity: 0;
      transform-origin: 50% 100%;
      transform: scale(0, 0);
    }
  }
  .rc-slider-tooltip {
    position: absolute;
    left: -9999px;
    top: -9999px;
    visibility: visible;
    box-sizing: border-box;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
    z-index: ${({ zIndex }) => zIndex};
  }
  .rc-slider-tooltip * {
    box-sizing: border-box;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  }
  .rc-slider-tooltip-hidden {
    display: none;
  }
  .rc-slider-tooltip-placement-top {
    padding: 4px 0 8px 0;
  }
  .rc-slider-tooltip-inner {
    font-family: Roboto;
    padding: 7px;
    min-width: 24px;
    font-weight: 500;
    height: 20px;
    font-size: 16px;
    line-height: 24px;
    letter-spacing: 0.2px;
    color: ${({ sType, theme }) => getTooltipColor(sType, theme)};
    text-align: center;
    text-decoration: none;
  }

  .rc-slider-tooltip-arrow {
    position: absolute;
    width: 0;
    height: 0;
    top: -4px;
    left: 50%;
  }
`;
