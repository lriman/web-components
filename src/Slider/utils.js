export const getRailBackgroundColor = (sType, theme) => {
  return theme.slider[sType].railBackgroundColor;
};

export const getTrackBackgroundColor = (sType, theme) => {
  return theme.slider[sType].trackBackgroundColor;
};

export const getHandleBorderColor = (sType, theme, isFocused, isHovered) => {
  if (isFocused) return theme.slider[sType].handle.borderColor.focused;
  if (isHovered) return theme.slider[sType].handle.borderColor.hovered;
  return theme.slider[sType].handle.borderColor.normal;
};

export const getHandleBackgroundColor = (sType, theme, isFocused, isHovered) => {
  if (isFocused) return theme.slider[sType].handle.backgroundColors.focused;
  if (isHovered) return theme.slider[sType].handle.backgroundColors.hovered;
  return theme.slider[sType].handle.backgroundColors.normal;
};

export const getSliderMarkColor = (sType, theme, topMark) => {
  if (topMark) return theme.slider[sType].markColor.accent;
  return theme.slider[sType].markColor.normal;
};

export const getSliderDotStyles = (sType, topMark, theme) => {
  if (topMark)
    return `
  position: absolute;    
  bottom: -9px;
  width: 2px;
  height: 25px;
  background-color: ${theme.slider[sType].dotBackgroundColor.accent};
  z-index: 2;
  `;

  return `
  position: absolute;    
  bottom: 0;
  width: 4px;
  height: 4px;
  background-color: ${theme.slider[sType].dotBackgroundColor.normal};
  border-radius: 50%;
  z-index: 0;
  `;
};

export const getTooltipColor = (sType, theme) => {
  return theme.slider[sType].tooltipColor;
};
