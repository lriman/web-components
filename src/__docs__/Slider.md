```jsx
const SliderExample = require('../__examples__/Slider.example').default;

const marksPercent = {
  0: '0',
  10: '10',
  20: '20',
  30: '30',
  40: '40',
  50: '50',
  60: '60',
  70: '70',
  80: '80',
  90: '90',
  99: '99%',
};

const marksSum = {
  0: '0',
  75000: '75 000 ₽',
};

const marksMonths = {
  6: '6',
  9: '9',
  12: '12',
  15: '15',
  18: '18',
  21: '21',
  24: '24 мес.',
};

<div className="omay-ds-block">
  <div className="omay-ds-block">
    <SliderExample id="testSlider1" min={0} max={100} />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSlider2" min={0} max={100} value={50} />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSlider3" min={0} max={99} value={50} marks={marksPercent} step={null} label="Первоначальный взнос" />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSlider4" min={0} max={99} value={50} marks={marksPercent} step={null} label="Первоначальный взнос" compact />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSlider5" min={0} max={75000} value={25000} marks={marksSum} step={2500} label="Первоначальны взнос" />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSlider6" min={6} max={24} value={12} marks={marksMonths} step={null} label="Срок кредита" />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSlider7" min={6} max={24} value={15} marks={marksMonths} step={null} label="Срок кредита" disabled />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSlider8" tooltip min={6} max={24} value={12} marks={marksMonths} step={null} label="Срок кредита" />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSlider8" topMark min={6} max={24} value={12} marks={marksMonths} step={null} label="Срок кредита" />
  </div>
</div>
```

### Slider Range
```jsx
const SliderExample = require('../__examples__/Slider.example').default;

const marksPercent = {
  0: '0',
  10: '10',
  20: '20',
  30: '30',
  40: '40',
  50: '50',
  60: '60',
  70: '70',
  80: '80',
  90: '90',
  99: '99%',
};

const marksSum = {
  0: '0',
  75000: '75 000 ₽',
};

const marksMonths = {
  6: '6',
  9: '9',
  12: '12',
  15: '15',
  18: '18',
  21: '21',
  24: '24 мес.',
};

<div className="omay-ds-block">
  <div className="omay-ds-block">
    <SliderExample id="testSliderRange1" range min={0} max={100} />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSliderRange2" range min={0} max={100} value={[25, 50]} />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSliderRange3" range min={0} max={99} value={[25, 50]} marks={marksPercent} step={null} label="Первоначальный взнос" />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSliderRange4" range min={0} max={99} value={[50, 99]} marks={marksPercent} step={null} label="Первоначальный взнос" compact />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSliderRange5" range min={0} max={75000} value={[0, 25000]} marks={marksSum} step={2500} label="Первоначальны взнос" />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSliderRange6" range min={6} max={24} value={[12, 24]} marks={marksMonths} step={null} label="Срок кредита" />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSliderRange7" range min={6} max={24} value={[15, 18]} marks={marksMonths} step={null} label="Срок кредита" disabled />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSliderRange8" tooltip range min={6} max={24} value={[12, 24]} marks={marksMonths} step={null} label="Срок кредита" />
  </div>
  <div className="omay-ds-block">
    <SliderExample id="testSliderRange8" topMark range min={6} max={24} value={[12, 24]} marks={marksMonths} step={null} label="Срок кредита" />
  </div>
</div>
```