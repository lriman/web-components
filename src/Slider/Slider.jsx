import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import RcSlider, { Range as RcSliderRange, createSliderWithTooltip } from 'rc-slider';

import { addDataAttributes } from '../_utils';

import { COMPONENTS, REFS } from './_constants';
import { StyledSlider, StyledSliderContent, StyledSliderField, StyledSliderLite } from './_style';

const RcSliderWithTooltip = createSliderWithTooltip(RcSlider);
const RcSliderRangeWithTooltip = createSliderWithTooltip(RcSliderRange);

export default class Slider extends PureComponent {
  static propTypes = {
    id: PropTypes.string.isRequired,
    min: PropTypes.number.isRequired,
    max: PropTypes.number.isRequired,
    marks: PropTypes.shape({}),
    step: PropTypes.number,
    value: PropTypes.oneOfType([PropTypes.number, PropTypes.arrayOf(PropTypes.number)]).isRequired,
    disabled: PropTypes.bool,
    range: PropTypes.bool,
    tooltip: PropTypes.bool,
    compact: PropTypes.bool,
    wideTrack: PropTypes.bool,
    topMark: PropTypes.bool,
    zIndex: PropTypes.number,
    type: PropTypes.oneOf(Object.values(REFS.TYPES)),
    onBeforeChange: PropTypes.func,
    onChange: PropTypes.func,
    onAfterChange: PropTypes.func,
    children: PropTypes.node,
    tipFormatter: PropTypes.func,
  };

  static defaultProps = {
    marks: undefined,
    step: undefined,
    disabled: false,
    range: false,
    tooltip: false,
    compact: false,
    wideTrack: false,
    topMark: false,
    zIndex: 6,
    type: REFS.TYPES.NORMAL,
    onBeforeChange: () => null,
    onChange: () => null,
    onAfterChange: () => null,
    children: null,
    tipFormatter: undefined,
  };

  handleBeforeChange = value => {
    const { id, onBeforeChange } = this.props;
    onBeforeChange({ id, value });
  };

  handleChange = value => {
    const { id, onChange } = this.props;
    onChange({ id, value });
  };

  handleAfterChange = value => {
    const { id, onAfterChange } = this.props;
    onAfterChange({ id, value });
  };

  render() {
    const {
      id,
      min,
      max,
      marks,
      step,
      value,
      disabled,
      range,
      tooltip,
      compact,
      type,
      children,
      tipFormatter,
      wideTrack,
      topMark,
      zIndex,
    } = this.props;

    const componentProps = {
      className: compact ? 'm-compact' : null,
      min,
      max,
      marks,
      step,
      value,
      disabled,
      onBeforeChange: this.handleBeforeChange,
      onChange: this.handleChange,
      onAfterChange: this.handleAfterChange,
    };

    return (
      <StyledSlider {...addDataAttributes({ component: COMPONENTS.GENERAL, id, disabled: disabled || null })}>
        <StyledSliderLite wideTrack={wideTrack} topMark={topMark} sType={type} zIndex={zIndex}>
          {children && <StyledSliderContent>{children}</StyledSliderContent>}
          <StyledSliderField isWithMarks={!!marks}>
            {tooltip && range && (
              <RcSliderRangeWithTooltip
                {...componentProps}
                allowCross={false}
                tipProps={{ placement: 'bottom', visible: true }}
                tipFormatter={tipFormatter}
              />
            )}
            {tooltip && !range && (
              <RcSliderWithTooltip
                {...componentProps}
                tipProps={{ placement: 'bottom', visible: true }}
                tipFormatter={tipFormatter}
              />
            )}
            {!tooltip && range && <RcSliderRange {...componentProps} allowCross={false} />}
            {!tooltip && !range && <RcSlider {...componentProps} />}
          </StyledSliderField>
        </StyledSliderLite>
      </StyledSlider>
    );
  }
}
