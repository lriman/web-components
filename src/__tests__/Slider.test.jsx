import React from 'react';
import TestRenderer from 'react-test-renderer';
import 'jest-styled-components';
import Slider from '../Slider';

describe('Slider', () => {
  it(`default`, () => {
    const wrapper = TestRenderer.create(<Slider id="slider" min={0} max={100} value={0} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it(`with marks`, () => {
    const wrapper = TestRenderer.create(
      <Slider id="slider" min={0} max={100} marks={{ 0: '0%', 50: '50%', 100: '100%' }} step={null} value={0} />,
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it(`with step`, () => {
    const wrapper = TestRenderer.create(<Slider id="slider" min={0} max={100} step={10} value={0} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it(`disabled`, () => {
    const wrapper = TestRenderer.create(<Slider id="slider" min={0} max={100} disabled value={0} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it(`with onChange`, () => {
    const wrapper = TestRenderer.create(
      <Slider id="slider" min={0} max={100} onChange={() => {}} value={0} />,
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it(`with children`, () => {
    const wrapper = TestRenderer.create(
      <Slider id="slider" min={0} max={100} onChange={() => {}} value={0}>
        <div>Slider</div>
      </Slider>,
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
