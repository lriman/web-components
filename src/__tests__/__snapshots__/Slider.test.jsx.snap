// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider default 1`] = `
.c0 {
  position: relative;
  box-sizing: border-box;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  padding: 0 10px 0 10px;
}

<div
  className="c0"
  data-component="Slider"
  data-disabled={null}
  data-id="slider"
>
  <div
    className="c1"
  >
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "0%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
  </div>
</div>
`;

exports[`Slider disabled 1`] = `
.c0 {
  position: relative;
  box-sizing: border-box;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  padding: 0 10px 0 10px;
}

<div
  className="c0"
  data-component="Slider"
  data-disabled={true}
  data-id="slider"
>
  <div
    className="c1"
  >
    <div
      className="rc-slider rc-slider-disabled"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "0%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={true}
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={null}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
  </div>
</div>
`;

exports[`Slider with children 1`] = `
.c0 {
  position: relative;
  box-sizing: border-box;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  margin-bottom: 4px;
}

.c2 {
  position: relative;
  box-sizing: border-box;
  padding: 0 10px 0 10px;
}

<div
  className="c0"
  data-component="Slider"
  data-disabled={null}
  data-id="slider"
>
  <div
    className="c1"
  >
    <div>
      Slider
    </div>
  </div>
  <div
    className="c2"
  >
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "0%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
  </div>
</div>
`;

exports[`Slider with marks 1`] = `
.c0 {
  position: relative;
  box-sizing: border-box;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  padding: 0 10px 16px 10px;
}

<div
  className="c0"
  data-component="Slider"
  data-disabled={null}
  data-id="slider"
>
  <div
    className="c1"
  >
    <div
      className="rc-slider rc-slider-with-marks"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "0%",
          }
        }
      />
      <div
        className="rc-slider-step"
      >
        <span
          className="rc-slider-dot rc-slider-dot-active"
          style={
            Object {
              "left": "0%",
            }
          }
        />
        <span
          className="rc-slider-dot"
          style={
            Object {
              "left": "50%",
            }
          }
        />
        <span
          className="rc-slider-dot"
          style={
            Object {
              "left": "100%",
            }
          }
        />
      </div>
      <div
        aria-disabled={false}
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      >
        <span
          className="rc-slider-mark-text rc-slider-mark-text-active"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "left": "0%",
              "msTransform": "translateX(-50%)",
              "transform": "translateX(-50%)",
            }
          }
        >
          0%
        </span>
        <span
          className="rc-slider-mark-text"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "left": "50%",
              "msTransform": "translateX(-50%)",
              "transform": "translateX(-50%)",
            }
          }
        >
          50%
        </span>
        <span
          className="rc-slider-mark-text"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "left": "100%",
              "msTransform": "translateX(-50%)",
              "transform": "translateX(-50%)",
            }
          }
        >
          100%
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider with onChange 1`] = `
.c0 {
  position: relative;
  box-sizing: border-box;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  padding: 0 10px 0 10px;
}

<div
  className="c0"
  data-component="Slider"
  data-disabled={null}
  data-id="slider"
>
  <div
    className="c1"
  >
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "0%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
  </div>
</div>
`;

exports[`Slider with step 1`] = `
.c0 {
  position: relative;
  box-sizing: border-box;
}

.c1 {
  position: relative;
  box-sizing: border-box;
  padding: 0 10px 0 10px;
}

<div
  className="c0"
  data-component="Slider"
  data-disabled={null}
  data-id="slider"
>
  <div
    className="c1"
  >
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "0%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
  </div>
</div>
`;
